lib/internal/repl.js

```diff
@@ -84,17 +99,24 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
     if (data) {
+      repl.history = data.split(/[\n\r]+/).slice(-repl.historySize);
+    } else if (oldHistoryPath) {
+      // Grab data from the older pre-v3.0 JSON NODE_REPL_HISTORY_FILE format.
+      repl._writeToOutput(
+          'Converting old JSON repl history to line-separated history.\n' +
+          `The new repl history file can be found at ${historyPath}.\n`);
```

Comment:
Might add `remove the NODE_REPL_HISTORY_FILE environment variable to stop this behavior.`